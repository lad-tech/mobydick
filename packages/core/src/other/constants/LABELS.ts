export const LABELS = {
  slider: 'slider',
  sliderLayoutLowThumb: 'sliderLayoutLowThumb',
  imageAvatar: 'imageAvatar',
  swipe: 'swipe',
  dotsAnimatedView: 'dotsAnimatedView',
  dotsScrollView: 'dotsScrollView',
  search: 'search',
  cancelSearch: 'cancelSearch',
  tab: 'tab',
  codeField: 'codeField',
  selector: 'selector',
  actionSheetsItem: 'actionSheetsItem',
  carousel: 'carousel',
  carouselItem: 'carouselItem',
  collapsed: 'collapsed',
  panelHeaderLeftView: 'panelHeaderLeftView',
  panelHeaderRightView: 'panelHeaderRightView',
  chatInputField: 'chatInputField',
  ratingStarButton: 'ratingStarButton',
};
