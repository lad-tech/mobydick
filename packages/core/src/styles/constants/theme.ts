import {TWeights} from '../../typography';
import {IDefaultTheme} from '../types';
import {getShadows} from '../shadows/getShadows';

import {defaultSpaces} from './defaultSpaces';
import {
  defaultAdditionalDarkColor,
  defaultAdditionalLightColor,
  defaultBannerDarkColor,
  defaultBannerLightColor,
  defaultBgDarkColor,
  defaultBgLightColor,
  defaultBorderDarkColor,
  defaultBorderLightColor,
  defaultCategoryDarkColor,
  defaultCategoryLightColor,
  defaultChartDarkColor,
  defaultChartLightColor,
  defaultCTADarkColor,
  defaultCTALightColor,
  defaultElementDarkColor,
  defaultElementLightColor,
  defaultIconDarkColor,
  defaultIconLightColor,
  defaultTextDarkColor,
  defaultTextLightColor,
} from './colors';

export enum CurrentTheme {
  light = 'light',
  dark = 'dark',
}
/** @deprecated  use defaultFont **/
export const defaultFonts: TWeights = {
  Regular: 'Inter-Regular',
  Medium: 'Inter-Medium',
  SemiBold: 'Inter-SemiBold',
  Bold: 'Inter-Bold',
};

export const defaultFont = 'Inter';

export const defaultTheme: IDefaultTheme = {
  currentTheme: CurrentTheme.light,
  colors: {
    [CurrentTheme.light]: {
      ...defaultTextLightColor,
      ...defaultElementLightColor,
      ...defaultIconLightColor,
      ...defaultBgLightColor,
      ...defaultCTALightColor,
      ...defaultBorderLightColor,
      ...defaultChartLightColor,
      ...defaultCategoryLightColor,
      ...defaultBannerLightColor,
      ...defaultAdditionalLightColor,
    },
    [CurrentTheme.dark]: {
      ...defaultTextDarkColor,
      ...defaultElementDarkColor,
      ...defaultIconDarkColor,
      ...defaultBgDarkColor,
      ...defaultCTADarkColor,
      ...defaultBorderDarkColor,
      ...defaultChartDarkColor,
      ...defaultCategoryDarkColor,
      ...defaultBannerDarkColor,
      ...defaultAdditionalDarkColor,
    },
  },
  spaces: {
    ...defaultSpaces,
  },
  fonts: defaultFonts,
  font: defaultFont,
  shadows: getShadows({
    spaces: defaultSpaces,
    currentTheme: CurrentTheme.light,
  }),
};
