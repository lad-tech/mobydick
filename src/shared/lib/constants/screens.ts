export enum SCREENS {
  Home = 'Home',
  Settings = 'Settings',
  Core = 'Core',
  Calendar = 'Calendar',
  Utils = 'Utils',
  Chart = 'Chart',

  Chat = 'Chat',
  Controls = 'Controls',
  CTA = 'CTA',
  Inputs = 'Inputs',
  Popups = 'Popups',
  Progress = 'Progress',
  Styles = 'Styles',
  TypographyAll = 'TypographyAll',
  TypographyLegacy = 'TypographyLegacy',
  Title = 'Title',
  Typography = 'Typography',
  Other = 'Other',
  Navbars = 'Navbars',

  LineChart = 'LineChart',
  BarChart = 'BarChart',

  KeyboardAware = 'KeyboardAware',
  KeyboardAwareScrollView = 'KeyboardAwareScrollView',
  KeyboardAwareScrollViewWithBottom = 'KeyboardAwareScrollViewWithBottom',

  KeyboardAwareScrollViewTabs= 'KeyboardAwareScrollViewTabsScreen',
  KeyboardAwareScrollViewWithTabs= 'KeyboardAwareScrollViewWithTabs',
  KeyboardAwareScrollViewWithBottomAndTabs = 'KeyboardAwareScrollViewWithBottomAndTabs',
  DragAndDrop = 'DragAndDrop',

  Markdown = 'Markdown',
}
