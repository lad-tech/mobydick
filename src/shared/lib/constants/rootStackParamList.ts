import { SCREENS } from "@shared/lib/constants/screens";

type IRootStackParamList = {
  [SCREENS.Home]: undefined,
  [SCREENS.Settings]: undefined

  [SCREENS.Core]: undefined
  [SCREENS.Calendar]: undefined
  [SCREENS.Utils]: undefined
  [SCREENS.Chart]: undefined

  [SCREENS.Chat]: undefined
  [SCREENS.Controls]: undefined
  [SCREENS.CTA]: undefined
  [SCREENS.Inputs]: undefined
  [SCREENS.Popups]: undefined
  [SCREENS.Progress]: undefined
  [SCREENS.Styles]: undefined
  [SCREENS.TypographyAll]: undefined
  [SCREENS.TypographyLegacy]: undefined
  [SCREENS.Title]: undefined
  [SCREENS.Typography]: undefined
  [SCREENS.Other]: undefined
  [SCREENS.Navbars]: undefined

  [SCREENS.LineChart]: undefined
  [SCREENS.BarChart]: undefined

  [SCREENS.KeyboardAware]: undefined
  [SCREENS.KeyboardAwareScrollView]: undefined
  [SCREENS.KeyboardAwareScrollViewWithBottom]: undefined

  [SCREENS.KeyboardAwareScrollViewTabs]: undefined
  [SCREENS.KeyboardAwareScrollViewWithTabs]: undefined
  [SCREENS.KeyboardAwareScrollViewWithBottomAndTabs]: undefined
  [SCREENS.DragAndDrop]: undefined
  [SCREENS.Markdown]: undefined
}

export default IRootStackParamList
