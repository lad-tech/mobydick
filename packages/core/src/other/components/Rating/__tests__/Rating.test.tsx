import {act, fireEvent, render} from '@testing-library/react-native';

import {Rating} from '../index';
import {LABELS} from '../../../constants';

describe('Rating', () => {
  test('render rating 0', () => {
    const onPress = jest.fn();
    const {toJSON} = render(
      <Rating count={0} setCurrentRate={onPress} currentRate={0} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  test('render rating 1', () => {
    const onPress = jest.fn();
    const {toJSON} = render(
      <Rating count={1} setCurrentRate={onPress} currentRate={1} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  test('render rating 2', () => {
    const onPress = jest.fn();
    const {toJSON} = render(
      <Rating count={2} setCurrentRate={onPress} currentRate={2} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  test('render rating 3', () => {
    const onPress = jest.fn();
    const {toJSON} = render(
      <Rating count={3} setCurrentRate={onPress} currentRate={3} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  test('render rating 4', () => {
    const onPress = jest.fn();
    const {toJSON} = render(
      <Rating count={4} setCurrentRate={onPress} currentRate={4} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  test('render rating 5', () => {
    const onPress = jest.fn();
    const {toJSON} = render(
      <Rating count={5} setCurrentRate={onPress} currentRate={5} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  test('change rating', () => {
    const onPress = jest.fn();
    const {toJSON, getByLabelText} = render(
      <Rating count={5} setCurrentRate={onPress} currentRate={2} />,
    );
    const setRatingButton = getByLabelText(`${LABELS.ratingStarButton}4`);

    act(() => {
      fireEvent.press(setRatingButton);
    });

    expect(setRatingButton.props.accessibilityState).toHaveProperty(
      'disabled',
      undefined,
    );
    expect(onPress).toHaveBeenCalledWith(4);

    expect(toJSON()).toMatchSnapshot();
  });
  test('change rating is disabled', () => {
    const onPress = jest.fn();
    const {toJSON, getByLabelText} = render(
      <Rating
        count={5}
        setCurrentRate={onPress}
        disabled={true}
        currentRate={5}
      />,
    );
    const setRatingButton = getByLabelText(`${LABELS.ratingStarButton}4`);

    act(() => {
      fireEvent.press(setRatingButton);
    });

    expect(setRatingButton.props.accessibilityState).toHaveProperty(
      'disabled',
      true,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
