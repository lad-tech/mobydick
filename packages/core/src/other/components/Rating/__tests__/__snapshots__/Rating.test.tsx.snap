// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating change rating 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityLabel="ratingStarButton0"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton1"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton2"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#cccccc"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton3"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#cccccc"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton4"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#cccccc"
        height={40}
        width={40}
      />
    </View>
  </View>
</View>
`;

exports[`Rating change rating is disabled 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityLabel="ratingStarButton0"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton1"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton2"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton3"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton4"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
</View>
`;

exports[`Rating render rating 0 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
/>
`;

exports[`Rating render rating 1 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityLabel="ratingStarButton0"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
</View>
`;

exports[`Rating render rating 2 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityLabel="ratingStarButton0"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton1"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
</View>
`;

exports[`Rating render rating 3 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityLabel="ratingStarButton0"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton1"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton2"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
</View>
`;

exports[`Rating render rating 4 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityLabel="ratingStarButton0"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton1"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton2"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton3"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
</View>
`;

exports[`Rating render rating 5 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityLabel="ratingStarButton0"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton1"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton2"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton3"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
  <View
    accessibilityLabel="ratingStarButton4"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <SvgMock
        fill="#ffd712"
        height={40}
        width={40}
      />
    </View>
  </View>
</View>
`;
